data-prepper-plugins/map-values-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/mapvalues/MapValues.java

```diff
@@ -0,0 +1,179 @@
+
+import org.apache.commons.lang3.Range;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.opensearch.dataprepper.expression.ExpressionEvaluator;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.processor.AbstractProcessor;
+import org.opensearch.dataprepper.model.processor.Processor;
+import org.opensearch.dataprepper.model.record.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.LinkedHashMap;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.opensearch.dataprepper.logging.DataPrepperMarkers.EVENT;
+
+
+@DataPrepperPlugin(name = "map-values", pluginType = Processor.class, pluginConfigurationType = MapValuesConfig.class)
+public class MapValues extends AbstractProcessor<Record<Event>, Record<Event>> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MapValues.class);
+    private final ExpressionEvaluator expressionEvaluator;
+    private final MapValuesConfig config;
+    private final LinkedHashMap<Range<Integer>, String> RANGES;
+    private final Map<String, String> MAP;
+
+    private final HashMap<String, String> PATTERNS;
+    @DataPrepperPluginConstructor
+    public MapValues(PluginMetrics pluginMetrics, final MapValuesConfig mapValuesConfig, final ExpressionEvaluator expressionEvaluator) {
+        super(pluginMetrics);
+        this.config = mapValuesConfig;
+        this.expressionEvaluator = expressionEvaluator;
+        // LOG.info("Entered mapvalues constructor");
+        MAP = new HashMap<>();
+        RANGES = new LinkedHashMap<>();
+        PATTERNS = config.getPatterns();
+        processMapField(mapValuesConfig.getMap());
+        parseFile(mapValuesConfig.getFilePath());
+
+    }
+
+    private void processMapField(Map<String, String> map){
+        for(Map.Entry<String, String> mapEntry : map.entrySet()){
+            String key = mapEntry.getKey();
+            if(key.contains(",")){
+                parseKeys(mapEntry);
+            }
+            else if(key.contains("-")){
+                parseRanges(mapEntry);
+            }
+            else{
+                MAP.put(key, map.get(key));
+            }
+        }
+    }
+
+    private void parseKeys(Map.Entry<String, String>  mapEntry){
+        String commaRegex = "([^,]+)(?:,|$)";
+        Pattern pattern = Pattern.compile(commaRegex);
+        Matcher matcher = pattern.matcher(mapEntry.getKey());
+        while (matcher.find()) {
+            String individualKey = matcher.group(1);
+            MAP.put(individualKey, mapEntry.getValue());
+        }
```

Comment:
This looks like a very complex way of doing. Is it not possible do something like
```
String[] keys = mapEntry.getKey().split(",");
```
It looks like the regex is also not handling commas inside strings anyways. What's the advantage of using regex pattern matching here instead of using string split?

Reply:
I was just trying something to see how it works. Will makeit simpler.